contrib/swagger.yaml

```diff
@@ -0,0 +1,439 @@
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Snippet'
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Bad request.
+    post:
+      summary: Create a new code snippet.
+      description: |
+        The Snippets endpoint creates a new code snippet.
+      parameters:
+        - name: snippet
+          in: body
+          schema:
+            $ref: '#/definitions/Snippet'
+      responses:
+        201:
+          schema:
+            $ref: '#/definitions/Snippet'
+          description: Created. A new Snippet has been successfully created.
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: Bad request.
+        401:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Unauthorized. Returned when a user tries to create a non-anonymous
+            snippet without authorizing first.
+        403:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Forbidden. Returned when an authorized user tries to create a snippet
+            on behalf of another user.
+
+  /snippets/{id}:
+    get:
+      summary: Retrieve information about the specific code snippet.
+      description: |
+        The Snippet endpoint returns information about the code snippet uniquely
+        identified by the given id value.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+      responses:
+        200:
+          description: A Snippet object
+          schema:
+            $ref: '#/definitions/Snippet'
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Bad request.
+        401:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Unauthorized. User tries to retrieve a private snippet without
+            authorizing first.
+        403:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Forbidden. User tries to retrieve a private snippet by another author.
+        404:
+          schema:
+            $ref: '#/definitions/Error'
+          description: Snippet not found.
+    patch:
+      summary: Create a new changeset of the snippet.
+      description: Create a new changeset of the snippet.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+        - name: content
+          in: body
+          schema:
+            $ref: '#/definitions/Changeset'
```

Comment:
Hm.. So `PATCH` on `/snippets/:id` must contain only 'changeset' body, correct?

I don't know what's better, but usually RESTful API says:

* use `PUT` when you need to replace resource content 'As Is' (in our case, snippet content, with implicit creation of new changeset)
* use `PATCH` when you need to replace only passed part of resource (in our case, snippet content, with, again, implict creation of new changeset in case `content` has been changed)

So we need to add `PUT` if we want to support, for instance, title renaming or something. That makes sense for `PATCH` either (if we want to have title renaming without creating a new changeset).

What do you think? I just a little bit confused.